### ISSUE TITLE
[WCAG 2.2 Draft Feedback] Success Criterion 3.2.6 Consistent Help (Level A)

### ISSUE THREAD

**Issue by dshoukry** [w3c/wcag#2709]
“Success Criterion 3.2.6 Consistent Help (Level A): If a web page contains any of the following help mechanisms, and those mechanisms are repeated on multiple web pages within a set of web pages, they occur in the same relative order to other page content, unless a change is initiated by the user:

- Human contact details;
- Human contact mechanism;
- Self-help option;
- A fully automated contact mechanism.”

Recommendation:
Most of our comments/proposals are requests to: add and edit the guidance to account for web applications with dynamic DOM structures, and simplifying some of the provided examples. 

Please find detailed specifics covered in our [Sep22 3.2.6 Consistent Help (A) Google Doc](https://docs.google.com/document/d/1WXROSroXObsJnOwWRlOXQddtoQfI_A7pnsxWIed3DMU/edit#)

**Comment by bruce-usab** [comment]
@dshoukry would you be willing to format your editorial suggestions as a Pull Request?  It is absolutely okay to decline!

**Comment by alastc** [comment]
@bruce-usab I remember discussing this with @dshoukry over email, and github PRs are not part of her normal work, the docs are the easiest for her to deal with.

**Comment by alastc** [comment]
Responding to the comments:

> As written, the SC is focused on how users navigate flat web pages and doesn't take into account web applications which may have very dynamic DOM structures, but consistent user flows for navigating to help resources.

There was a lot of discussion about how this would apply to dynamic webapps. From looking at examples, it was thought that these apps are very likely to pass as they have a template that applies across various views. 

The definitions of page and set-of-web-pages are difficult to apply in those scenarios, but essentially if it has the same template, same purpose & authors, it would be considered a 'set of web pages'. If the template has the human contact mechanism, it would pass.

The DOM structure is obviously dynamic, but it did not appear that things would be dynamically added that affect the help links. For example, Google docs has help at the end of the menu at the top. So long as the things above it (e.g. title, share button, account link) don't get moved underneath, that would pass. (And assuming the things underneath don't get moved above the menu.)

Is there an example that you consider problematic?


> Suggest removing this [mention of serialisation] as this may cause false positives when the DOM order changes but still conceptually makes sense to the user and seems consistent in overall page ordering.

The SC is constructed in this way as it is the closest thing to 'consistent location' that we can use. Given that layouts can change at different sizes / zoom levels / orientations, considering the serialised view is the only way to get consistent results. It is not 1:1 with layout consistency, but in practice it is close.

> Is it possible to simplify this paragraph? [Of the second note.] For example, if the guideline is primarily concerned with same page variations, perhaps this information should be pushed up front followed by a caveat?

This has been updated since you took the document snapshot, it is now:
"For this Success Criterion, the same relative order can be thought of as how the content is ordered when the page is serialized. The visual position of a help mechanism is likely to be consistent across pages for the same page variation (e.g., CSS break-point). The user can initiate a change, such as changing the page's zoom or orientation, which may trigger a different page variation. This criterion is concerned with relative order across pages displayed in the same page variation (e.g., same zoom level and orientation)."
